examples/connection_pools/src/main.rs

```diff
@@ -0,0 +1,198 @@
+#![feature(plugin, decl_macro)]
+#![plugin(rocket_codegen)]
+
+extern crate rocket;
+extern crate rocket_contrib;
+
+use std::io;
+use std::ops::Deref;
+use rocket::http::Status;
+use rocket::request::{self, FromRequest};
+use rocket::{Outcome, Request, State};
+use rocket::fairing::{Fairing, AdHoc};
+use rocket_contrib::databases::{self, DatabaseConfig, Poolable, diesel, postgres, r2d2, r2d2_postgres};
+
+// The two annotations would be part of the proc-macro crate to generate the
+// FromRequest implementation for the specified DB type.
+// #[derive(DbConn)]
```

Comment:
I'm thinking: do we even need the `derive` here? We should probably just have a `database` attribute. We don't have a `DbConn` trait; what we're actually deriving is `FromRequest`, strangely enough. Or, we should have a `DbConn` trait with a blanket impl of `FromRequest` for all `T: DbConn`.

Reply:
I'm starting to agree with that. Semantically, we're lying to the user if we continue with the `#[derive(DbConn)]` syntax we originally had planned without creating a `DbConn` trait.

I'm ambivalent on the approach we decide to take here. At the very least, just using the `database` attribute is the minimal change. What are the pros and cons of having a blanket impl of `FromRequest` for all `T: DbConn`?